Ensure the log messages from operators during parsing go somewhere

ashb opened the pull request:
While investigating #26599 and the change from AIP-45, I noticed that these warning messages weren't new! The only thing that was new was that we started seeing them.

This is because the logger for BaseOperator and all subclasses is `airflow.task.operators`, and the `airflow.task` logger is not configured (with `set_context()`) until we have a TaskInstance, so it just dropped all messages on the floor!

This changes it so that log messages are propagated to parent loggers by default, but when we configure a context (and thus have a file to write to) we stop that. A similar change was made for the `airflow.processor` (but that is unlikely to suffer the same fate)

ashb committed:
Ensure the log messages from operators during parsing go somewhere

While investigating #26599 and the change from AIP-45, I noticed that
these warning messages weren't new! The only thing that was new was that
we started seeing them.

This is because the logger for BaseOperator and all subclasses is
`airflow.task.operators`, and the `airflow.task` logger is not
configured (with `set_context()`) until we have a TaskInstance, so it
just dropped all messages on the floor!

This changes it so that log messages are propagated to parent loggers by
default, but when we configure a context (and thus have a file to write
to) we stop that. A similar change was made for the `airflow.processor`
(but that is unlikely to suffer the same fate)

kaxil commented on airflow/utils/log/logging_mixin.py:
```diff
@@ -179,15 +182,15 @@ def set_context(logger, value):
-                set_context(value)
-        if _logger.propagate is True:
-            _logger = _logger.parent
+                if set_context(value) is DISABLE_PROPOGATE:
```
Should we just chain this "if" with the "if" above?

```python
if set_context and set_context(value) is DISABLE_PROPOGATE:
```

ashb replied:
Can do, sure

mertozb replied:
Generally the visibility of warnings may be added into cfg as a section like deprecation or dependency warnings so that we can able to path around them.

ashb commented on airflow/utils/log/logging_mixin.py:
```diff
@@ -179,15 +182,15 @@ def set_context(logger, value):
-        if _logger.propagate is True:
-            _logger = _logger.parent
+                if set_context(value) is DISABLE_PROPOGATE:
+                    logger.propagate = False
```
```suggestion
                logger.propagate = False
```

ashb commented on airflow/utils/log/logging_mixin.py:
```diff
@@ -179,15 +182,15 @@ def set_context(logger, value):
             # Not all handlers need to have context passed in so we ignore
             # the error when handlers do not have set_context defined.
             set_context = getattr(handler, 'set_context', None)
             if set_context:
```
```suggestion
            if set_context and set_context(value) is DISABLE_PROPOGATE:
```

ashb committed:
Update airflow/utils/log/logging_mixin.py

ashb committed:
Update airflow/utils/log/logging_mixin.py

ashb committed:
fixup! Ensure the log messages from operators during parsing go somewhere

ashb committed:
Give a real row count value so logs don't fail

The ArangoDB sensor test was logging a mock object, which previously was
getting dropped before emitting, but with this change now fails with
"Mock is not an integer" when attempting the  `%d` intepolation.

To avoid making the mock overly specific (`arangodb_client_for_test.db.`
`return_value.aql.execute.return_value.count.return_value`!) I have
changed the test to mock the hook entirely (which is already tested)
